src/Entities/Chunk.php

```diff
@@ -0,0 +1,386 @@
+            $this->data     = unserialize( $data->data );
+            $this->start    = $data->start;
+            $this->end      = $data->end;
+            $this->duration = $data->duration;
+        }
+
+        $this->is_data_fetched = true;
+    }
+
+    /**
+     * Get the chunk ID
+     *
+     * @return int
+     */
+    public function get_chunk_id(): int {
+        return $this->chunk_id;
+    }
+
+    /**
+     * Get the action ID
+     *
+     * @return int
+     */
+    public function get_action_id(): int {
+        if ( ! $this->is_data_fetched ) {
+            $this->fetch_data();
+        }
+        return $this->action_id;
+    }
+
+    /**
+     * Get the name
+     *
+     * @return string
+     */
+    public function get_name(): string {
+        if ( ! $this->is_data_fetched ) {
+            $this->fetch_data();
+        }
+        return $this->name;
+    }
+
+    /**
+     * Get the group
+     *
+     * @return string
+     */
+    public function get_group(): string {
+        if ( ! $this->is_data_fetched ) {
+            $this->fetch_data();
+        }
+        return $this->group;
+    }
+
+    /**
+     * Get the status
+     *
+     * @return string
+     */
+    public function get_status(): string {
+        if ( ! $this->is_data_fetched ) {
+            $this->fetch_data();
+        }
+        return $this->status;
+    }
+
+    /**
+     * Get the data
+     *
+     * @return array<mixed>
+     */
+    public function get_data(): array {
+        if ( ! $this->is_data_fetched ) {
+            $this->fetch_data();
+        }
+        return $this->data;
+    }
+
+    /**
+     * Get the start time
+     *
+     * @return string
+     */
+    public function get_start(): string {
+        if ( ! $this->is_data_fetched ) {
+            $this->fetch_data();
+        }
+        return $this->start;
+    }
+
+    /**
+     * Get the end time
+     *
+     * @return string
+     */
+    public function get_end(): string {
+        if ( ! $this->is_data_fetched ) {
+            $this->fetch_data();
+        }
+        return $this->end;
+    }
+
+    /**
+     * Get the duration in seconds
+     *
+     * @return float
+     */
+    public function get_duration(): float {
+        if ( ! $this->is_data_fetched ) {
+            $this->fetch_data();
+        }
+
+        // If duration is not yet calculated, calculate it
+        if ( empty( $this->duration ) && ! empty( $this->end ) && ! empty( $this->start ) ) {
+            return $this->calculate_duration();
+        }
+
+        return (float) $this->duration;
+    }
+
+    /**
+     * Get formatted duration in human-readable format
+     *
+     * @return string
+     */
+    public function get_formatted_duration(): string {
+        $duration = $this->get_duration();
+
+        if ( 0.0 === $duration ) {
+            return '0s';
+        }
+
+        // Format with microseconds precision
+        return sprintf( '%.6fs', $duration );
```

Comment:
This does not belong to the class. Since php hast a number_format function there is no need for abstraction. Please remove